hack/lib/golang.sh

```diff
@@ -397,7 +397,7 @@ kube::golang::build_binaries_for_platform() {
         "${nonstatics[@]:+${nonstatics[@]}}"
     fi
     if [[ "${#statics[@]}" != 0 ]]; then
-      CGO_ENABLED=0 go install -installsuffix cgo "${goflags[@]:+${goflags[@]}}" \
+      CGO_ENABLED=1 go install -installsuffix cgo "${goflags[@]:+${goflags[@]}}" \
```

Comment:
I wouldn't expect to need this change. Assuming there are platform dependencies, add build tags to the PAM password impl to limit dependencies to the PAM lib to linux (or whatever other platforms support it) and stub the impl for other platforms